src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,39 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int minfigures = 5;
+    private final double defaultvalue = 10;
+    private final String defaultcolor = "white";
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int choice = random.nextInt(minfigures);
+        switch (choice) {
+            case 0:
+                return new Square(random.nextDouble() * 10 + 1, colorSupplier.getRandomColor());
```

Comment:
this expression
random.nextDouble() * 10 + 1
used many times
extract it to separate method with proper naming, like getRandomSide() or something like that